src/snowflake/cli/plugins/object/commands.py

```diff
@@ -92,3 +104,45 @@ def describe(
     return QueryResult(
         ObjectManager().describe(object_type=object_type, name=object_name)
     )
+
+
+def _parse_object_definition(object_definition: List[str]) -> Dict[str, Any]:
+    import json
+
+    def _parse_list_to_dict(object_definition: List[str]) -> Dict[str, Any]:
+        payload = {}
+        for item in object_definition:
+            try:
+                key, value = item.split("=", 1)
+                # try to parse non-string values
+                payload[key] = json.loads(value)
+            except json.JSONDecodeError:
+                payload[key] = value
+            except ValueError:
+                raise ClickException(f"expected key=value format, got {item}")
+        return payload
+
+    if len(object_definition) != 1:
+        return _parse_list_to_dict(object_definition)
+
+    # for list of length 1, prefer json error message
+    try:
+        return json.loads(object_definition[0])
+    except json.JSONDecodeError as json_err:
+        try:
+            return _parse_list_to_dict(object_definition)
+        except ClickException:
+            raise json_err
+
+
+@app.command(name="create", requires_connection=True, hidden=True)
+def create(
+    object_type: str = ObjectArgument,
+    object_definition: List[str] = ObjectDefinitionArgument,
```

Comment:
Have you considered separating the inputs into two: 
* `object_definition: dict` - where we use callback to parse the data
* `object_attirutes: List[str]` - where we support key=value specification

Won't that help with logic complexity?

Reply:
Yes, but the usage would be more painful. My thoughts:
If we wanted to have two arguments, both need to be optional, which means we have to use `typer.Option`, which require flags. So usage would be `-D key=value -D key=value` or `--json '{"name": "my_db"}'`

My personal preference is simpler usage (without multiple `-D`), but I'm open for discussion

Reply:
Fixed to `snow init [ATTRIBUTES] [--json DEFINITION]` format